src/components/SeedInfo/SeedInfoViews/FungalShifts.tsx

```diff
@@ -1,83 +1,110 @@
-import React, { useContext } from 'react';
-import classnames from 'classnames';
+import { useContext } from 'react';
 import { Stack, Form, Tooltip, OverlayTrigger } from 'react-bootstrap';
 
 import GameInfoProvider from '../../../services/SeedInfo/infoHandler';
 import { capitalize } from '../../../services/helpers';
 import { FungalInfoProvider } from '../../../services/SeedInfo/infoHandler/InfoProviders/Fungal';
 import { AlchemyConfigContext } from '../../AlchemyConfigContext';
 import { useTranslation } from 'react-i18next';
-import classNames from 'classnames';
 import { useMaterialFavorite } from './helpers';
+import { MaterialInfoProvider } from '../../../services/SeedInfo/infoHandler/InfoProviders/Material';
 
 const Flask = () => {
 	return <b className="text-info">Flask</b>;
 };
 
+interface IFungalMaterialProps {
+	materials: Map<string, string>
+	direction: boolean
+	isFlask: boolean
+	isFavorite // TODO: add type 
+	showId // TODO: add type 
+}
+
+const FungalMaterial = (props: IFungalMaterialProps) => {
+	const key = props.direction ? "to" : "from"
+
+	/*
+	Var 'props.materials' may contain multiple materials with the same display name.
+	This is because some materials like 'Flammable Gas' have static variants that are
+	pre-rendered in a scene. Static materials have their own material ID.
+
+	These materials should not be displayed twice.
+
+	To solve this, materials will be sorted in a new `Map` where the key is the material
+	name acts as the key and the material ids are stored in the value as an array.
+	*/
+
+	const materialsByName = new Map()
+	props.materials.forEach((matName, matId) => {
+		const ids = materialsByName.get(matName) ?? []
+		ids.push(matId)
+		materialsByName.set(matName, ids)
+	});
+
+	// The logic that returns a JSX component requires an `Array`
+	const materialsByNameArray = Array.from(materialsByName)
+
+	return(
+		<Stack>
+			{props.isFlask && <Flask />}
+			{materialsByNameArray.map(([matName, matIds]) => {
```

Comment:
I think context already gives hints that name is for the materials, so it might be more concise to just use name and ids

Reply:
Resolved by Resolved by rename